crypto/crypto-derive/src/hasher.rs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+/// Converts a camel-case string to snake-case
+pub fn camel_to_snake(text: &str) -> String {
+    let mut out = String::with_capacity(text.len());
+    let mut first = true;
+    text.chars().for_each(|c| {
+        if !first && c.is_uppercase() {
```

Comment:
qq: do we accept non-Ascii identifiers?

Reply:
No, they can't be passed as an input to this macro (https://doc.rust-lang.org/reference/identifiers.html).